src/App.js

```diff
@@ -17,48 +17,41 @@ import NavSection from "components/NavSection";
 import Route from "components/Routes/Route";
 import { buildLocationsSet } from "./utils/buildLocationsSet";
 import { getUniqueVenues } from "./utils/getUniqueVenues";
+import { useDataApi } from "./hooks/useDataApi";
 
 function App() {
   const [footerVisible, setFooterVisible] = useState(true);
+  const { data, fetching, callApi } = useDataApi();
   const { isGeolocationAvailable, coords, mode } = useContext(GeoContext);
-  const {
-    setData,
-    setLocations,
-    selectedLocation,
-    setSelectedIndex,
-    setFetchingData,
-  } = useContext(AppContext);
+  const { setData, setLocations, setSelectedIndex } = useContext(AppContext);
 
-  useEffect(() => {
-    setFetchingData(true);
-
-    let URL;
-    if ((isGeolocationAvailable && mode === "geo") || mode === "postcode") {
-      if (coords) {
-        const {latitude, longitude} = coords
-        URL = `${BASE_PROVIDERS_LAMBDA}?&coords=${latitude},${longitude}`;
-      }
-    } else {
-      URL = ALL_PROVIDERS_LAMBDA
+  // to be refactored
+  let URL;
+  if ((isGeolocationAvailable && mode === "geo") || mode === "postcode") {
+    if (coords) {
+      const {latitude, longitude} = coords
+      URL = `${BASE_PROVIDERS_LAMBDA}?&coords=${latitude},${longitude}`;
     }
+  } else {
+    URL = ALL_PROVIDERS_LAMBDA
+  }
 
-    fetch(URL)
-      .then((response) => response.json())
-      .then(async (data) => {
-        setFetchingData(false);
-        // first result in spreadsheet is null.
-        if (mode == null) {	
-          data.shift();	
-        }
-        
-        const uniqueVenuesArray = getUniqueVenues(data)
-        setData(uniqueVenuesArray);
+  useEffect(() => {
+    callApi(URL);
+    if (data) {
+      // first result contains null
+      if (mode === null) {
+        data.shift()
```

Comment:
I don't like how we're getting the data in the hook, then bring it back here and doing more manipulation on it. It feels messy and unecessary. Can the data.shift and unique stuff be done on the lambda function? 

there's also an issue with the coordinates; if you use the chrome debugger to put 3 breakpoints - line 16 and line 19, useDataAPI, and one on line 48 of App.js - you'll see fetch is called 3 times. It then calls `setData`, once with the full array (before the browser has given the app the coords) and then again with the geolocated results. But then in App.js it doesn't seem to called `setData` with the geo results.